src/main/java/frc/robot/VisionFieldLayoutRecognizer.java

```diff
@@ -0,0 +1,88 @@
+package frc.robot;
+
+import java.util.ArrayList;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import frc.robot.AutonomousPath.VisionPathName;
+
+
+/*
+ * VisionFieldLayoutRecognizer
+ * 
+ * Version 1.0
+ *
+ * 02/27/2021
+ * 
+ * EnTech 
+ */
+
+ public class VisionFieldLayoutRecognizer
+    implements PixyFieldPoseResolver{
+
+     
+    
+    public VisionFieldLayoutRecognizer() {
+      
+    }
+
+     public AutonomousPath detectPose(ArrayList<FieldVisionInput> arraylistFieldVisionInput) {
+        
+      AutonomousPath objectAutonomousPath;
+      FieldVisionInput objectFieldVisionInput;
+      DistanceRange objectDistanceRange;
+
+      double dblFocalLength = 11.33;//pixels = 3mm
+      double dblWidth = 7; //inches
+      double dblDistance = 0; 
+      double dblAverageDistance = 0;
+      double[] dblArrPixels = new double[3];
+
+      
+      objectAutonomousPath = new AutonomousPath();
+      objectDistanceRange = new DistanceRange();
+
+      // This is the input array that will hold the values provided by the pixie cam 
+      // regarding the pixels from each ball seen
+      // I am commenting the below code on 02/27/2021
+      // I am going to calculate the pixels seen by multiplying
+      // the height and width that the pixie camera returns to us
+      // when we make the call to getFieldVisionInput above
+      //// double[] dblArrPixels = {5.0,13.46,10.31};
+      
+  
+      // Just for refernce the formula to calculate the focal length
+      //dblFocalLength = (dblPixels*dblDistance)/dblWidth;
+
+      if (arraylistFieldVisionInput.size() >  0 )   
+      {
+        for(int count = 0; count < arraylistFieldVisionInput.size(); count++){
+
+          objectFieldVisionInput = arraylistFieldVisionInput.get(count);
+
+          dblArrPixels[count] = objectFieldVisionInput.getObjectHeight() * 
+                                  objectFieldVisionInput.getObjectWidth();
+                                  
+        }
+      }
+      
+      // For loop that adds the distance of the three balls
+      for (int i = 0; i < dblArrPixels.length; i++) {
+         dblDistance = dblDistance + (dblWidth*dblFocalLength)/dblArrPixels[i];
+  
+      }
+      SmartDashboard.putNumber("Distance: ", dblDistance);
```

Comment:
as we dicussed, i'd like to not mix all this smartdashboard stuff in here. Why? because this class cant be unit tested without running on the roborio!!!! (Smartdashboard call wont work)